packages/quantic/force-app/main/default/lwc/quanticNotifications/quanticNotifications.js

```diff
@@ -56,8 +56,10 @@ export default class QuanticNotifications extends LightningElement {
     this.unsubscribe?.();
   }
 
-  updateState() {
-    this.notifyTriggerState = this.notifyTrigger.state;
+  updateState(updatedNotifications) {
```

Comment:
I don't really like what's being done with this function,  as in I don't like that this function can now support having a parameter or not, 

before this change this function had one clear job, updating the component state when the Headless state changes, clear and concise now it's not only that, and I think we can achieve the same result in a simpler way.


Plus I don't think it's a good idea to kinda separate the Headless state from the component state, with this logic added in this PRs it will be possible to have for example in the Headless state: `['notification 1', 'notification 2']` and in the component state just ['notification 1'].

What would be much better I think than removing data is to augment data with a key that indicates whether the notification is visible or not. for example the state in Headless would be `['notification 1', 'notification 2']`, and the state in the component would be:`[{value: 'notification 1', visible: true}, {value:'notification 2', visible: false}]`

I think this way of doing would yield a simpler code, easy to read and easier to understand,
It would be something like this:

```javascript
  updateState() {
    this.notifications = this.notifyTrigger?.state?.notifications.map((notification, index) => ({
      value: notification,
      id: index.toString(),
      visible: true,
    })) ?? [];
  
    this.ariaLiveNotificationsRegion.dispatchMessage(
      this.notifyTriggerState?.notifications.reduce(
        (value, notification, index) => {
          return `${value} Notification ${index + 1}: ${notification}`;
        },
        ''
      )
    );
  }

  handleNotificationClose(event) {
    const currentNotificationId = event.currentTarget.dataset.id;
    this.notifications = this.notifications.map((notification) => {
      if (notification.id === currentNotificationId) {
        return { ...notification, visible: false }
      }
      return notification
    });
  }

then in the HTML you only display a notification only when visible is true.
This way I think at first glance you can understand everything going on with this component.
```

Reply:
yeah good point!